controllers/UserController.js

```diff
@@ -0,0 +1,20 @@
+const { models } = require('../models');
+const { hashPassword } = require('../helpers');
+
+class UserController {
+  static async createUser(req, res) {
+    try {
+      const password = await hashPassword(req.body.password);
+      req.body.password = password;
+      const result = await models.User.create(req.body);
+      return res.status(201).json({ result });
+    } catch (err) {
+      if (err.code === 11000) {
+        return res.status(409).json({ message: 'username exists already' });
+      }
```

Comment:
I'm worried this error handling is scalable. As there are many `err.code`s which we will not be able to cover. Maybe we can just return the `err.message`?